nixos/modules/services/networking/pppd.nix

```diff
@@ -65,7 +76,10 @@ in
 
     mkEtc = peerCfg: {
       name = "ppp/peers/${peerCfg.name}";
-      value.text = peerCfg.config;
+      value.text = ''
+        ${peerCfg.config}
+        ${if peerCfg.waitUntilReady then "up_sdnotify" else ""}
```

Comment:
this could be a `lib.optionalString`

Reply:
Fixed. Thanks for the suggestion, and sorry for the delay (was traveling).